signer/tests/integration/transaction_coordinator.rs

```diff
@@ -1347,30 +1383,54 @@ async fn sign_bitcoin_transaction() {
             bitcoin_blocks: ReceiverStream::new(receiver),
             horizon: 10,
         };
-
+        let counter = start_count.clone();
         tokio::spawn(async move {
             counter.fetch_add(1, Ordering::Relaxed);
             block_observer.run().await
         });
     }
 
-    while start_count.load(Ordering::SeqCst) < 9 {
+    while start_count.load(Ordering::SeqCst) < 12 {
         tokio::time::sleep(Duration::from_millis(10)).await;
     }
 
-    faucet.generate_blocks(1);
-
-    // 5. Once they are all running, signal that DKG should be run. We
-    //    signal them all because we do not know which one is the
-    //    coordinator.
-
-    tokio::time::sleep(Duration::from_secs(3)).await;
-
-    let mut shares: EncryptedDkgShares = Faker.fake_with_rng(&mut rng);
-    for (_, db, _, _) in signers.iter() {
-        shares = db.get_latest_encrypted_dkg_shares().await.unwrap().unwrap();
-    }
+    // =========================================================================
+    // Step 4 - Wait for DKG
+    // -------------------------------------------------------------------------
+    // - Once they are all running, generate a bitcoin block to kick off
+    //   the database updating process.
+    // - After they have the same view of the canonical bitcoin blockchain,
+    //   the signers should all participate in DKG.
+    // =========================================================================
+    let chain_tip: BitcoinBlockHash = faucet.generate_blocks(1).pop().unwrap().into();
+
+    // We first need to wait for bitcoin-core to send us all the
+    // notifications so that we are up to date with the chain tip.
+    let db_update_futs = signers
+        .iter()
+        .map(|(_, db, _, _)| testing::storage::wait_for_chain_tip(db, chain_tip));
+    futures::future::join_all(db_update_futs).await;
+    tokio::time::sleep(Duration::from_secs(2)).await;
 
+    // Now we wait for DKG to successfully complete. For that we just watch
+    // the dkg_shares table. Also, we need to get the signers' scriptPubKey
+    // so that we can make a donation, and get the party started.
+    let dkg_futs = signers
+        .iter()
+        .map(|(_, db, _, _)| testing::storage::wait_for_dkg(db));
+    futures::future::join_all(dkg_futs).await;
```

Comment:
suggestion: i'd just `use` these, `Join_all`, `try_join_all` etc. are pretty clear what they do/where they come from

Reply:
My rule of thumb is to not import functions unless using the full path leads to really long code lines. Then again, `use`ing some common functions like these leads to increased code readability so maybe there should be some exceptions. Plus these are tests 🤔

Reply:
I'm on the fence, so I went with no action.